HealthcareSystem/EventSourcingService/Service/PatientSchedulingEventService/EventStorePatientEndSchedulingService.cs

```diff
@@ -0,0 +1,206 @@
+﻿using EventSourcingService.DTO;
+using EventSourcingService.Model;
+using EventSourcingService.Model.Enum;
+using EventSourcingService.Repository;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace EventSourcingService.Service
+{
+    public class EventStorePatientEndSchedulingService : IEventStorePatientEndSchedulingService
```

Comment:
Some of the functions in this class could be merged into one, I think? I haven't looked that carefully but I feel like you do a lot of calculating the same statistics for successful and unsuccessful, and it's two separate methods instead of one that takes a parameter.